### PR TITLE
PRESIDECMS-656

### PR DESCRIPTION
**Pull request by john5onCheng**
Ensures object picker is able to filter from related simple value field.

### CODE REVIEW

**Comment by john5onCheng** [comment]
Hi Dom, I've revert previous changes and apply the recommended approach.

**Comment by DominicWatson** [comment]
This looks about right - I tried it out locally though and couldn't get it to work. There are a few things here that I'd refactor too. Let's take a look together as soon as reasonable.

**Comment by john5onCheng** [comment]
Hi Dom, I've found the error that you mentioned and have get it fixed. Please let me know which line you would like to refactor.

File: system/handlers/admin/DataManager.cfc [hunk 1]
```diff
@@ -194,11 +194,24 @@
 
 			_checkPermission( argumentCollection=arguments, key="read", object=objectName );
 
+			var filterByField = rc.filterByField ?: "";
+			var extraFilters  = [];
+
+			loop list=filterByField index="indexField"{
```
Comment (DominicWatson) [inline system/handlers/admin/DataManager.cfc]: `indexField` is not `var'd` here and I would avoid looping a list - especially using cfloop syntax. Prefer this:

```
var filterByFields = ListToArray( rc.filterByField ?: "" );
for( var filterByField in filterByFields ) {
    // ...
}
```

File: system/assets/js/admin/presidecore/preside.uber.select.js [hunk 2]
```diff
@@ -365,10 +365,21 @@
 				if ( this.local_options[i].empty ) { this.local_options.splice( i, 1 ); }
 			}
 
+			var prefetch_url = this.form_field.getAttribute( "data-prefetch-url" );
+			var remote_url   = this.form_field.getAttribute( "data-remote-url" );
```
Comment (DominicWatson) [inline system/assets/js/admin/presidecore/preside.uber.select.js]: in our *js*, we'd like `var` definitions at the _top_ of functions and using the style:

```
var variable1 = true
  , variable2 = false
  , variable3, variable 4; // not yet set vars
```

These vars should be moved up aroud line 355 (and declared in the list style).

File: system/assets/js/admin/presidecore/preside.uber.select.js [hunk 3]
```diff
@@ -500,7 +510,31 @@
...
+					$uberSelect.setup_search_engine();
+				});
+			}
+		};
+
+		UberSelect.prototype.setup_filter = function() {
+
+			this.filter         = "" ;
+
+			var filterBy        = this.form_field.getAttribute( "data-filter-by" );
```
Comment (DominicWatson) [inline system/assets/js/admin/presidecore/preside.uber.select.js]: All the `var` definitions should be moved to the top of the function here:

```
var filterBy        = this.form_field.getAttribute( "data-filter-by" )
  , filterByField   = this.form_field.getAttribute( "data-filter-by-field" )
  , filterByValue;

this.filter       = "";
this.filter_field = $( "input[name='" + filterBy + "']" );

filterByValue = this.filter_field.val();
if ( typeof filterByValue !== "undefined" ) {
// ...
}
```

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Looking good, please see review comments re points that need refactoring.

**Comment by john5onCheng** [comment]
hi Dom, requested changes has been made

**Review by DominicWatson — APPROVED** [review]

**Comment by DominicWatson** [comment]
Perfect, testing now.

**Pull request merged**